project-koku/koku
# Refine and cleanup local dev setup

## Description

- Rename all containers to have a `koku-` prefix
- Remove prefix from service name for less typing
- Use dashes instead of underscores for container names to align with default default in Compose v2
- Updates default `.env.example` file to use correct values and add missing ones
- Don't use hotfix version of `minio` since there is no ARM container for hotfix tags
- Update some deprecated `mc` commands

## Testing

1. Checkout Branch
1. Stop and remove all existing koku containers, base image, and build cache

    ```
    docker stop $(docker ps --filter name=koku -q)
    docker rm $(docker ps  -a --filter name=koku -q)
    docker rmi koku_base
    docker buildx prune -af
    ```

1. Remove existing Minio data. 

    ```
    rm -rf .trino/parquet_data
    ```

1. Source the new values in `.env.example` or update them in existing `.env` file.

1. Start new containers

    ```
    make docker-up
    ```

1. Verify services work as expected

    ```
    make docker-logs
    ```
